project_redss/auto_code_show_messages.py

```diff
@@ -72,16 +71,22 @@ def auto_code_show_messages(cls, user, data, icr_output_dir, coda_output_dir):
                     not_noise, plan.raw_field, cls.SENT_ON_KEY, plan.id_field, {}, f
                 )
 
-        # Randomly select some messages to export for ICR
-        # TODO: ICR
-        icr_messages = ICRTools.generate_sample_for_icr(not_noise, cls.ICR_MESSAGES_COUNT, random.Random(0))
-
-        # Output ICR data to a CSV file
-        # run_id_key = "{} (Run ID) - {}".format(cls.VARIABLE_NAME, cls.FLOW_NAME)
-        # raw_text_key = "{} (Text) - {}".format(cls.VARIABLE_NAME, cls.FLOW_NAME)
-        # IOUtils.ensure_dirs_exist_for_file(icr_output_path)
-        with open(icr_output_dir, "w") as f:
-            f.write("")
-            # TracedDataCSVIO.export_traced_data_iterable_to_csv(icr_messages, f, headers=[run_id_key, raw_text_key])
+        # Output messages for ICR
+        IOUtils.ensure_dirs_exist(icr_output_dir)
+        for plan in DatasetSpecification.RQA_CODING_PLANS:
+            rqa_messages = []
+            for td in not_noise:
+                # This test works because the only codes which have been applied at this point are TRUE_MISSINGs.
+                # If any other coding is done above, this test will need to change.
+                if plan.coded_field not in td:
+                    rqa_messages.append(td)
+
+            icr_messages = ICRTools.generate_sample_for_icr(rqa_messages, cls.ICR_MESSAGES_COUNT, random.Random(0))
+
+            icr_output_path = path.join(icr_output_dir, f"{plan.coda_filename}.csv")
```

Comment:
{coda_filename}.csv?

Is this a legacy from the days when coda took csv files?